### PR TITLE
[SDK] Deployment of Extension Pattern contracts

### PR DESCRIPTION
**Pull request by kumaryash90**

### CODE REVIEW

File: packages/sdk/src/evm/common/any-evm-utils.ts [hunk 1]
```diff
@@ -944,10 +995,33 @@ export async function encodeConstructorParamsForImplementation(
         }
 
         return nativeTokenWrapperAddress;
+      } else if (p.name && p.name.includes("linkTokenAddress")) {
+        const chainId = (await provider.getNetwork()).chainId;
+        let linkTokenAddress = getLinkTokenByChainId(chainId);
+
+        if (linkTokenAddress === ethers.constants.AddressZero) {
+          throw new Error(
+            "Chainlink LINK token not found on this network. Please provide as a custom param if available.",
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/common/any-evm-utils.ts:1004]: oof another tricky one for any EVM support. gonna need to restrict the chains at the publish level for packVRF

File: packages/sdk/src/evm/common/any-evm-utils.ts [hunk 2]
```diff
@@ -978,6 +1052,21 @@ export async function encodeConstructorParamsForImplementation(
           deploymentPresets["Forwarder"] = deploymentInfo;
         }
 
+        return deploymentInfo.transaction.predictedAddress;
+      } else if (p.name && p.name.includes("entrypoint")) {
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/common/any-evm-utils.ts:1056]: is this for smart accounts?

File: packages/sdk/src/evm/common/plugin.ts [hunk 3]
```diff
@@ -189,11 +190,80 @@ export async function getMetadataForPlugins(
           ).map((fetchedMetadata) => fetchedMetadata.compilerMetadata);
         }
       } catch {}
+    } else {
+      throw new Error(
+        "No implementation found for Router. Can't fetch plugin metadata",
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/common/plugin.ts:195]: are you sure we want to throw here? is not enough to return empty array?

File: packages/sdk/src/evm/common/plugin.ts [hunk 4]
```diff
@@ -189,11 +190,80 @@ export async function getMetadataForPlugins(
...
+              extensionUris.map(async (uri) => {
+                return fetchAndCacheDeployMetadata(uri, storage);
+              }),
+            )
+          ).map((fetchedMetadata) => fetchedMetadata.compilerMetadata);
+        }
+      } catch {}
+    } else {
+      throw new Error(
+        "No implementation found for Router. Can't fetch extension metadata",
```
Comment (joaquim-verges) [inline packages/sdk/src/evm/common/plugin.ts:260]: same here

File: packages/sdk/src/evm/common/any-evm-utils.ts (hunk 2 above)
Comment (kumaryash90) [reply packages/sdk/src/evm/common/any-evm-utils.ts:1056]: Yes, Entrypoint seems to be the only dependency there, for constructor input.

File: packages/sdk/src/evm/common/plugin.ts (hunk 3 above)
Comment (kumaryash90) [reply packages/sdk/src/evm/common/plugin.ts:195]: I added it temporarily, till we change the publish flow for dynamic contracts.
Right now we need at least one implementation, to be able to resolve the default extensions.

The empty array will eventually throw when encoding constructor params, but the messaging won't be clear.

**Comment by joaquim-verges** [comment]
closing for now @kumaryash90

**Pull request closed without merging**